cmd/outputs_cloud.go

```diff
@@ -0,0 +1,171 @@
+	conf, warn, err := cloudapi.GetConsolidatedConfig(
+		test.derivedConfig.Collectors[builtinOutputCloud.String()],
+		gs.Env,
+		"", // Historically used for -o cloud=..., no longer used (deprecated).
+		test.derivedConfig.Options.Cloud,
+		test.derivedConfig.Options.External,
+	)
+	if err != nil {
+		return err
+	}
+
+	if warn != "" {
+		gs.Logger.Warn(warn)
+	}
+
+	// If not, we continue with some validations and the creation of the test run.
+	if err := validateRequiredSystemTags(test.derivedConfig.Options.SystemTags); err != nil {
+		return err
+	}
+
+	if !conf.Name.Valid || conf.Name.String == "" {
+		scriptPath := test.source.URL.String()
+		if scriptPath == "" {
+			// Script from stdin without a name, likely from stdin
+			return errors.New("script name not set, please specify K6_CLOUD_NAME or options.cloud.name")
+		}
+
+		conf.Name = null.StringFrom(filepath.Base(scriptPath))
+	}
+	if conf.Name.String == "-" {
+		conf.Name = null.StringFrom(defaultTestName)
+	}
+
+	thresholds := make(map[string][]string)
+	for name, t := range test.derivedConfig.Thresholds {
+		for _, threshold := range t.Thresholds {
+			thresholds[name] = append(thresholds[name], threshold.Source)
+		}
+	}
+
+	et, err := lib.NewExecutionTuple(
+		test.derivedConfig.Options.ExecutionSegment,
+		test.derivedConfig.Options.ExecutionSegmentSequence,
+	)
+	if err != nil {
+		return err
+	}
+	executionPlan := test.derivedConfig.Options.Scenarios.GetFullExecutionRequirements(et)
+
+	duration, testEnds := lib.GetEndOffset(executionPlan)
+	if !testEnds {
+		return errors.New("tests with unspecified duration are not allowed when outputting data to k6 cloud")
+	}
+
+	if conf.MetricPushConcurrency.Int64 < 1 {
+		return fmt.Errorf("metrics push concurrency must be a positive number but is %d",
+			conf.MetricPushConcurrency.Int64)
+	}
+
+	if conf.MaxTimeSeriesInBatch.Int64 < 1 {
+		return fmt.Errorf("max allowed number of time series in a single batch must be a positive number but is %d",
+			conf.MaxTimeSeriesInBatch.Int64)
+	}
+
+	var testArchive *lib.Archive
+	if !test.derivedConfig.NoArchiveUpload.Bool {
+		testArchive = test.initRunner.MakeArchive()
+	}
+
+	testRun := &cloudapi.TestRun{
+		Name:       conf.Name.String,
+		ProjectID:  conf.ProjectID.Int64,
+		VUsMax:     int64(lib.GetMaxPossibleVUs(executionPlan)),
+		Thresholds: thresholds,
+		Duration:   int64(duration / time.Second),
+		Archive:    testArchive,
+	}
+
+	logger := gs.Logger.WithFields(logrus.Fields{"output": builtinOutputCloud.String()})
+
+	apiClient := cloudapi.NewClient(
+		logger, conf.Token.String, conf.Host.String, consts.Version, conf.Timeout.TimeDuration())
+
+	response, err := apiClient.CreateTestRun(testRun)
+	if err != nil {
+		return err
+	}
+
+	if response.ConfigOverride != nil {
+		logger.WithFields(logrus.Fields{"override": response.ConfigOverride}).Debug("overriding config options")
+		conf = conf.Apply(*response.ConfigOverride)
```

Comment:
This lint error likely catches an issue.

I guess we might need to set this options specifically for the cloud output when it actually starts

Reply:
Yeah, I think in this case (when there's cloud config overrides) we still need to previous code you suggested to remove, at least for now, as it is what the Cloud output will use 🤔